File: sdk/src/main/java/eu/darken/porter/sdk/PorterServerInfo.java
package eu.darken.porter.sdk;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/** What the server on one connection reported about itself when that connection attached. */
public final class PorterServerInfo {

    /** Which protocol the reported numbers are on the scale of. */
    @NonNull
    public final PorterBackend backend;

    /** The version this connection's server reported at attach; 0 when it reported none. */
    public final int version;

    /** The Shizuku patch level, where the backend reports one; null where it has none. */
    @Nullable
    public final Integer patchVersion;

    PorterServerInfo(@NonNull PorterBackend backend, int version, @Nullable Integer patchVersion) {
        this.backend = backend;
        this.version = version;
        this.patchVersion = patchVersion;
    }
}

File: sdk/src/main/java/eu/darken/porter/sdk/PorterServerInfo.kt
package eu.darken.porter.sdk

/** What the server on one connection reported about itself when that connection attached. */
public data class PorterServerInfo(
    /** Which protocol the reported numbers are on the scale of. */
    val backend: PorterBackend,
    /** The version this connection's server reported at attach; 0 when it reported none. */
    val version: Int,
    /** The Shizuku patch level, where the backend reports one; null where it has none. */
    val patchVersion: Int?,
)
